pkg/controller/master/backup/backup_metadata.go

```diff
@@ -140,7 +143,47 @@ func (h *MetadataHandler) OnBackupTargetChange(_ string, setting *harvesterv1.Se
 		return nil, nil
 	}
 
-	return nil, nil
+	return h.renewBackupTarget(setting)
+}
+
+func (h *MetadataHandler) shouldRefresh(setting *harvesterv1.Setting, refreshIntervalInSeconds int64) bool {
+	if setting.Annotations == nil {
+		return true
+	}
+
+	var err error
+	lastTime := time.Unix(0, 0)
+	currentTime := time.Now()
+	if setting.Annotations[util.AnnotationLastRefreshTime] != "" {
+		lastTime, err = time.Parse(time.RFC3339, setting.Annotations[util.AnnotationLastRefreshTime])
+		if err != nil {
+			logrus.WithError(err).Errorf("failed to parse last refresh time")
+			lastTime = time.Unix(0, 0)
+			return true
+		}
+	}
+	if getBackupTargetHash(setting.Value) == setting.Annotations[util.AnnotationHash] &&
+		(refreshIntervalInSeconds == 0 ||
```

Comment:
did some one notice that, when refreshIntervalInSeconds ==0,

h.settings.EnqueueAfter(setting.Name, lastTime.Add(time.Duration(refreshIntervalInSeconds)*time.Second).Sub(currentTime))

What value will be used to EnqueueAfter? a negative value?

Looks like the `refreshIntervalInSeconds == 0` should be processed indepedently, instead of call `EnqueueAfter`.

Reply:
Yes, good catch. Updated in https://github.com/harvester/harvester/pull/7684.